README.md

```diff
@@ -1,135 +1,285 @@
+  const res = await auth.getApplicationDefault();
+  let client = res.credential;
 
-authFactory.getApplicationDefault(function(err, authClient) {
-  if (err) {
-    console.log('Authentication failed because of ', err);
-    return;
+  // The createScopedRequired method returns true when running on GAE or a local developer
+  // machine. In that case, the desired scopes must be passed in manually. When the code is
+  // running in GCE or a Managed VM, the scopes are pulled from the GCE metadata server.
+  // See https://cloud.google.com/compute/docs/authentication for more information.
+  if (client.createScopedRequired && client.createScopedRequired()) {
+    // Scopes can be specified either as an array or as a single, space-delimited string.
+    const scopes = ['https://www.googleapis.com/auth/cloud-platform'];
+    client = client.createScoped(scopes);
+  }
+  return {
+    client: client,
+    projectId: res.projectId
   }
-  if (authClient.createScopedRequired && authClient.createScopedRequired()) {
-    var scopes = ['https://www.googleapis.com/auth/cloud-platform'];
-    authClient = authClient.createScoped(scopes);
+}
+
+main();
+```
+
+#### OAuth2 client
+
+This client comes with an [OAuth2][oauth] client that allows you to retrieve an access token and refreshes the token and retry the request seamlessly if you also provide an `expiry_date` and the token is expired. The basics of Google's OAuth2 implementation is explained on [Google Authorization and Authentication documentation][authdocs].
+
+In the following examples, you may need a `CLIENT_ID`, `CLIENT_SECRET` and `REDIRECT_URL`. You can find these pieces of information by going to the [Developer Console][devconsole], clicking your project --> APIs & auth --> credentials.
+
+For more information about OAuth2 and how it works, [see here][oauth].
+
+##### A complete OAuth2 example
+
+Let's take a look at a complete example.
+
+``` js
+const OAuth2Client = require('google-auth-library').OAuth2Client;
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const opn = require('opn');
+
+// Download your OAuth2 configuration from the Google
+const keys = require('./keys.json');
+
+/**
+ * Start by acquiring a pre-authenticated oAuth2 client.
+ */
+async function main() {
+  try {
+    const oAuth2Client = await getAuthenticatedClient();
+    // Make a simple request to the Google Plus API using our pre-authenticated client. The `request()` method
+    // takes an AxiosRequestConfig object.  Visit https://github.com/axios/axios#request-config.
+    const url = 'https://www.googleapis.com/plus/v1/people?query=pizza';
+    const res = await oAuth2Client.request({url})
+    console.log(res.data);
+  } catch (e) {
+    console.error(e);
   }
+  process.exit();
+}
+
+/**
+ * Create a new OAuth2Client, and go through the OAuth2 content
+ * workflow.  Return the full client to the callback.
+ */
+async function getAuthenticatedClient() {
```

Comment:
style nit: use `async` only when you need to use an `await`.

Reply:
It's using await in that method to call `getToken()`

Reply:
The function that calls `getToken` is already async. This one doesn't need to be.

Reply:
Ah, good point.  Updated!